eman2/convert/convert.py

```diff
@@ -414,3 +511,137 @@ def calculatePhaseShift(ampcont):
+        # values: '{"coverage": 1.0, "score": 2.0, "xform.align3d": {"matrix": [...]}}'
+        import re
+        match = re.search(r'(\d+)\)$', key)
+        if not match:
+            continue
+        particleIndex = int(match.group(1))
+        coverage = values.get("coverage")
+        score = values.get("score")
+        alignMatrix = values.get("xform.align3d", {}).get("matrix")
+
+        if coverage and score and alignMatrix:
+            customParticleParams = dict(
+                coverage=coverage,
+                score=score,
+                alignMatrix=alignMatrix
+            )
+            output[particleIndex] = customParticleParams
+
+    return output
+
+
+def updateSetOfSubTomograms(inputSetOfSubTomograms, outputSetOfSubTomograms, particlesParams):
+    """Update a set of subtomograms from a template and copy attributes coverage/score/transform"""
+
+    def updateSubTomogram(subTomogram, index):
+        particleParams = particlesParams.get(index)
+        if not particleParams:
+            raise Exception("Could not get params for particle %d" % index)
+        setattr(subTomogram, 'coverage', Float(particleParams["coverage"]))
+        setattr(subTomogram, 'score', Float(particleParams["score"]))
+        # Create 4x4 matrix from 4x3 e2spt_sgd align matrix and append row [0,0,0,1]
+        am = particleParams["alignMatrix"]
+        angles = numpy.matrix([am[0:3], am[4:7], am[8:11], [0, 0, 0]])
+        samplingRate = outputSetOfSubTomograms.getSamplingRate()
+        shift = numpy.matrix([am[3] * samplingRate, am[7] * samplingRate, am[11] * samplingRate, 1])
+        matrix = numpy.concatenate((angles, shift.T), axis=1)
+        subTomogram.setTransform(Transform(matrix))
+
+    outputSetOfSubTomograms.copyItems(inputSetOfSubTomograms,
+                                      updateItemCallback=updateSubTomogram,
+                                      itemDataIterator=itertools.count(0))
+
+
+def setCoords2Jsons(setTomograms, setCoords, path):
+    for tomo in setTomograms.iterItems():
+        coords = []
+        for coor in setCoords.iterCoordinates():
+            if pwutils.removeBaseExt(tomo.getFileName()) == pwutils.removeBaseExt(coor.getVolName()):
+                coords.append([coor.getX(), coor.getY(), coor.getZ(), "manual", 0.0, 0])
+
+        coordDict = {"boxes_3d": coords,
+                     "class_list": {"0": {"boxsize": setCoords.getBoxSize(), "name": "particles_00"}}
+                     }
+        tomoBasename = pwutils.removeBaseExt(tomo.getFileName())
+        if "__" in tomoBasename:
+            fnInputCoor = '%s_info.json' % tomoBasename.split("__")[0]
+        else:
+            fnInputCoor = 'extra-%s_info.json' % tomoBasename
+        pathInputCoor = pwutils.join(path, fnInputCoor)
+        if coords:
+            writeJson(coordDict, pathInputCoor)
+
+def jsons2SetCoords(protocol, setTomograms, outPath):
+    from tomo.objects import SetOfCoordinates3D
```

Comment:
3d in the name too?

Reply:
Done https://github.com/scipion-em/scipion-em-eman2/pull/64/commits/3ab25a6fd8a7029dfe0412c7b336e06d62560ed7